pkg/api/controllers/transaction_controller.go

```diff
@@ -192,40 +193,49 @@ func (ctl *TransactionController) RevertTransaction(c *gin.Context) {
 		return
 	}
 
-	respondWithData[*core.Transaction](c, http.StatusOK, tx)
+	respondWithData[core.Transaction](c, http.StatusOK, tx)
 }
 
 func (ctl *TransactionController) PostTransactionMetadata(c *gin.Context) {
 	l, _ := c.Get("ledger")
 
 	var m core.Metadata
 	if err := c.ShouldBindJSON(&m); err != nil {
-		panic(err)
+		ResponseError(c, ledger.NewValidationError("invalid metadata format"))
+		return
 	}
 
 	txId, err := strconv.ParseUint(c.Param("txid"), 10, 64)
 	if err != nil {
-		c.AbortWithStatus(http.StatusBadRequest)
+		ResponseError(c, ledger.NewValidationError("invalid transaction ID"))
 		return
 	}
 
-	if err := l.(*ledger.Ledger).SaveMeta(c.Request.Context(), core.MetaTargetTypeTransaction, txId, m); err != nil {
```

Comment:
By design, adding metadata on a non existent transaction could exists before that.
Maybe it is ok but we have to check.